modules/alloy/src/main/ts/ephox/alloy/system/InternalAttachment.ts

```diff
@@ -53,15 +79,52 @@ const replaceChildren = (component: AlloyComponent, newChildren: AlloyComponent[
     } else {
       attach(component, childComp);
     }
-    component.syncComponents();
   });
+  component.syncComponents();
+};
+
+const virtualReplaceChildren = (component: AlloyComponent, newSpecs: AlloySpec[], buildNewChildren: (newSpecs: AlloySpec[]) => AlloyComponent[]): void => {
+  // Detach all existing child components that aren't premade specs
```

Comment:
Maybe a further comment here along the lines of: "we don't want to fire detachedFromDom and attachedToDom again for a premade that has just had its position in the children moved around, so we only detach if initially if we aren't a premade. Premades will be detached later, but only if they are no longer in the child list.

Reply:
Added what you had here verbatim basically (minor wording fix)